tensorflow/python/training/learning_rate_decay.py

```diff
@@ -84,3 +86,49 @@ def exponential_decay(learning_rate, global_step, decay_steps, decay_rate,
     if staircase:
       p = math_ops.floor(p)
     return math_ops.mul(learning_rate, math_ops.pow(decay_rate, p), name=name)
+
+def piecewise_constant(x, boundaries, values, name=None):
+  """ Piecewise constant from boundaries and interval values.
+  
+  Example: use a learning rate that's 1.0 for the first 100000 steps, 0.5
+    for steps 100001 to 110000, and 0.1 for any additional steps.
+  
+  ```python
+  global_step = tf.Variable(0, trainable=False)
+  boundaries = [100000, 110000]
+  values = [1.0, 0.5, 0.1]
+  learning_rate = tf.train.piecewise_constant(global_step, boundaries, values)
+  
+  # Later, whenever we perform an optimization step, we increment global_step.
+  ```
+  
+  Args:
+    x: A 0-D scalar Tensor. Must be one of the following types: `float32`,
+    `float64`, `uint8`, `int8`, `int16`, `int32`, `int64`.
+    boundaries: A list of `float`s or `int`s with strictly increasing entries.
+    values: A list of `float`s or `int`s that specifies the values for the
+      intervals defined by `boundaries`. (It should therefore have one more
+      element than `boundaries`.
```

Comment:
How about accepting a list of scalar tensors for each element of `boundaries` and `values`? We can then use `ops.convert_n_to_tensor()` internally, to avoid converting the scalars so many times, and reduce the number of nodes this function adds to the graph.

Note that I think the elements of `boundaries` have to have the same dtype as `x`, and all elements of `values` must have the same type.